### PR TITLE
chore(deps): upgrade ESLint to v9

### PR DESCRIPTION
**Pull request by ReidyT**

### CODE REVIEW

**Review by pyphilia — COMMENTED** [review]
LGTM, I think @spaenleh will know better^^

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+      'react/jsx-filename-extension': [
+        'warn',
+        {
+          extensions: ['.tsx'],
+        },
+      ],
+
+      'react/function-component-definition': [
+        2,
```
Comment (spaenleh) [inline eslint.config.mjs:158]:
```suggestion
        "error",
```

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    ignores: [
+      '**/build',
+      '**/public',
+      '**/coverage',
+      '**/node_modules',
+      '.yarn/.cache',
+      '**/.husky',
+      '**/.nyc_output',
+      '**/.yarn',
+      '**/commitlint.config.js',
```
Comment (spaenleh) [inline eslint.config.mjs:33]:
```suggestion
      '**/commitlint.config.cjs',
```

**Review by spaenleh — APPROVED** [review]
Looks good ! Just a small fix.

One last thing you should do also is in VSCode set the `ESlint: Use flat config` to true.

**Pull request merged**